features/login/impl/src/main/kotlin/io/element/android/features/login/impl/login/LoginHelper.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+package io.element.android.features.login.impl.login
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import io.element.android.features.login.impl.DefaultLoginUserStory
+import io.element.android.features.login.impl.error.ChangeServerError
+import io.element.android.features.login.impl.screens.confirmaccountprovider.LoginFlow
+import io.element.android.features.login.impl.screens.createaccount.AccountCreationNotSupported
+import io.element.android.features.login.impl.web.WebClientUrlForAuthenticationRetriever
+import io.element.android.libraries.architecture.AsyncData
+import io.element.android.libraries.architecture.runCatchingUpdatingState
+import io.element.android.libraries.matrix.api.auth.MatrixAuthenticationService
+import io.element.android.libraries.matrix.api.auth.OidcPrompt
+import io.element.android.libraries.oidc.api.OidcAction
+import io.element.android.libraries.oidc.api.OidcActionFlow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class LoginHelper @Inject constructor(
+    private val oidcActionFlow: OidcActionFlow,
+    private val authenticationService: MatrixAuthenticationService,
+    private val defaultLoginUserStory: DefaultLoginUserStory,
+    private val webClientUrlForAuthenticationRetriever: WebClientUrlForAuthenticationRetriever,
+) {
+    private lateinit var loginFlowAction: MutableState<AsyncData<LoginFlow>>
+
+    val loginFlow: AsyncData<LoginFlow>
+        get() = loginFlowAction.value
+
+    @Composable
+    fun Start() {
```

Comment:
Maybe this could behave like `collectAsState`, directly returning the `loginFlowAction` current value so you don't have to expose it separately?